test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +178,298 @@ int suiteTearDown( int numFailures )
+
+    MQTT_Disconnect_ExpectAndReturn( &mqttAgentContext, MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Disconnect( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_TRUE( returnFlags.endLoop );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_Ping() works as intended.
+ */
+void test_MQTTAgentCommand_Ping( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Ping_ExpectAndReturn( &mqttAgentContext, MQTTSuccess );
+    mqttStatus = MQTTAgentCommand_Ping( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Connect() success case.
+ */
+void test_MQTTAgentCommand_Connect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentConnectArgs_t connectInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Connect_IgnoreAndReturn( MQTTSuccess );
+    MQTTAgent_ResumeSession_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Connect( &mqttAgentContext, &connectInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Connect() failure case.
+ */
+void test_MQTTAgentCommand_Connect_failure( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentConnectArgs_t connectInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Connect_IgnoreAndReturn( MQTTBadParameter );
+
+    mqttStatus = MQTTAgentCommand_Connect( &mqttAgentContext, &connectInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_terminate() works as intended.
+ */
+void test_MQTTAgentCommand_terminate( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTStatus_t mqttStatus;
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    Command_t command = { 0 };
+    AgentMessageInterface_t messageInterface = { 0 };
+
+    messageInterface.pMsgCtx = &globalMessageContext;
+    messageInterface.send = stubSend;
+    messageInterface.recv = stubReceive;
+    messageInterface.releaseCommand = stubReleaseCommand;
+    messageInterface.getCommand = stubGetCommand;
+
+    command.pCommandCompleteCallback = stubCompletionCallback;
+    mqttAgentContext.agentInterface = messageInterface;
```

Comment:
You'd save two lines by just not having `messageInterface`
```suggestion

    mqttAgentContext.agentInterface.pMsgCtx = &globalMessageContext;
    mqttAgentContext.agentInterface.send = stubSend;
    mqttAgentContext.agentInterface.recv = stubReceive;
    mqttAgentContext.agentInterface.releaseCommand = stubReleaseCommand;
    mqttAgentContext.agentInterface.getCommand = stubGetCommand;

    command.pCommandCompleteCallback = stubCompletionCallback;
```